datamap/lucene/src/main/java/org/apache/carbondata/datamap/lucene/LuceneDataMapFactoryBase.java

```diff
@@ -153,8 +153,8 @@ public DataMapWriter createWriter(Segment segment, String shardName) {
   }
 
   @Override
-  public DataMapRefresher createRefresher(Segment segment, String shardName) {
-    return new LuceneDataMapRefresher(getCarbonTable().getTablePath(), dataMapName,
+  public DataMapBuilder createBuilder(Segment segment, String shardName) {
```

Comment:
A question that does not relate to this PR: what's the relationship between a `segment` and a `shard`? (in traditional carbondata table storage and in other table storage, such as external table/ hive-partition-like table...)

Reply:
For a managed table (traditional carbondata table) a shard is an index generated by same write task while loading, inside one segment. 
For external table, shard concept is the same, there are segments and shards inside each partition folder. There is a segment file in meta folder to indicate the segment and shard path.